File: Algorithms/C++/sum_of_digits.cpp
#include <iostream>

using namespace std;

int sum_of_digits(int n)
{
    int sum = 0;
    while (n > 0)
    {
        sum += n % 10;
        n /= 10;
    }

    return sum;
}

int main()
{
    int number = 12345;

    int result = sum_of_digits(number);
    cout << "Sum of digit: " << result << endl;

    return 0;
}

File: Algorithms/Javascript/sumOfDigits.js
const sumOfDigits = (number) => {
    if (number === 0) {
        return 0;
    }

    let result = 0;
    let num = number;

    while (num > 0) {
        result += num % 10;
        num = Math.floor(num / 10);
    }

    return result;
}

console.log(sumOfDigits(123)); // 6
